src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -1,132 +1,162 @@
-	}
-	
-}
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Inventory;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptConfig;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+enum ReplacementTypes {
+	ALL,
+	FIRST,
+	LAST
+}
+
+@Name("Replace")
+@Description("Replaces all occurrences of a given text with another text and returns the replaced text.")
+@Examples({"replace \"<item>\" in {textvar} with \"%item%\"",
+	"replace every \"&\" with \"§\" in line 1",
+	"# The following acts as a simple chat censor:",
+	"on chat:",
+	"\treplace all \"kys\", \"idiot\" and \"noob\" with \"****\" in the message",
+	"#If you'd like a more advanced chat censor you can use regex:",
+	"on chat:",
+	"\treplace all \"(i|1)d(i|1)(o|0)(t|7)\" with \"****\" in message # which will block 'idiot' but also words in which a vowel is replaced with a number",
+	"replace all stone and dirt in player's inventory and player's top inventory with diamond"})
+@Since("2.0, 2.2-dev24 (replace in muliple strings and replace items in inventory), INSERT VERSION (replace first, replace last, case sensitivity and regex support)")
+public class EffReplace extends Effect {
+	static {
+		Skript.registerEffect(EffReplace.class,
+			"[(4¦case-sensitive)] replace (1¦first|2¦last|0¦all|every|) %strings% with %string% in %strings%",
+			"[(4¦case-sensitive)] replace (1¦first|2¦last|0¦all|every|) %strings% in %strings% with %string%",
+			"regex replace (1¦first|2¦last|0¦all|every|) %strings% with %string% in %strings%",
+			"replace (all|every|) %itemtypes% in %inventories% with %itemtype%",
+			"replace (all|every|) %itemtypes% with %itemtype% in %inventories%"
+		);
+	}
+	
+	private String type = "ALL";
```

Comment:
Still have a ``String`` instead of the ``ReplacementType`` enum. So this line should looks like
```java
private ReplacementTypes type = ReplacementType.ALL;
```

Reply:
Okay, thanks, also where should I put the file for that enum?

Reply:
Probably in ``ch.njol.skript.util``